### PR TITLE
Move server into go routine and add test

### PR DESCRIPTION
**Pull request by bn222**

### CODE REVIEW

**Comment by bn222** [comment]
@SalDaniele @wizhaoredhat PTAL

File: daemon/hostdaemon.go [hunk 1]
```diff
@@ -51,21 +56,37 @@ func NewHostDaemon(vsp plugin.VendorPlugin) *HostDaemon {
 	}
 }
 
+func (d *HostDaemon) ensureConnected() {
+	if d.conn != nil {
+		return
+	}
+	retryPolicy := `{
+		"methodConfig": [{
+		  "waitForReady": false,
```
Comment (wizhaoredhat) [inline daemon/hostdaemon.go:65]: Add a comment about waitForReady. It is not obvious what this does. We should say the effect is this makes Dial nonblocking and returns immediately.

File: daemon/daemon_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package main_test
```
Comment (wizhaoredhat) [inline daemon/daemon_test.go:1]: Why do we need a main test? Why can't this test be called vendorplugin_test?

File: dpu-cni/pkgs/cniserver/cniserver_test.go [hunk 3]
```diff
@@ -35,6 +35,26 @@ func processRequest(request *cnitypes.Request) (*current.Result, error) {
...
+	cmdArgs := &skel.CmdArgs{
+		ContainerID: "fakecontainerid",
+		Netns:       "fakenetns",
+		IfName:      "fakeeth0",
+		Args:        "",
+		Path:        "fakepath",
+		StdinData:   []byte(cniConfig),
+	}
+	os.Clearenv()
+	os.Setenv("CNI_COMMAND", "ADD")
```
Comment (wizhaoredhat) [inline dpu-cni/pkgs/cniserver/cniserver_test.go:49]: We should add macros for ADD and DEL and CHECK

File: daemon/daemon_test.go (hunk 2 above)
Comment (bn222) [reply daemon/daemon_test.go:1]: We're actually testing the daemon, and since daemon is in main, I've used main_test.

**Comment by bn222** [comment]
I've addressed all your concerns in separate commits. I will squash them when you have reviewed the changes.

**Pull request merged**